libcaf_core/caf/detail/parser/add_ascii.hpp

```diff
@@ -29,11 +33,24 @@ namespace parser {
 // @returns `false` on an overflow, otherwise `true`.
 // @pre `isdigit(c) || (Base == 16 && isxdigit(c))`
 template <int Base, class T>
-bool add_ascii(T& x, char c) {
+bool add_ascii(T& x, char c, enable_if_tt<std::is_integral<T>, int> u = 0) {
+  CAF_IGNORE_UNUSED(u);
+  ascii_to_int<Base, T> f;
+  auto y = f(c);
+  if (x > ((std::numeric_limits<T>::max() / Base) - y)
+            + (std::numeric_limits<T>::max() % Base))
```

Comment:
This does not work in all cases. Here are a few test cases that should fail, but don't and give a wrong address as a result:

```
  should_fail("260.12.12.12");
  should_fail("261.12.12.12"); 
  should_fail("262.12.12.12");
  should_fail("263.12.12.12");
  should_fail("264.12.12.12");
  should_fail("270.12.12.12");
  should_fail("271.12.12.12"); 
  should_fail("272.12.12.12"); 
  should_fail("273.12.12.12"); 
  should_fail("280.12.12.12"); 
  should_fail("281.12.12.12");  
  should_fail("282.12.12.12"); 
  should_fail("291.12.12.12"); 
  should_fail("300.12.12.12");
```

Reply:
Oh, and why is the `BASE` part of the check. It is only important for converting the ascii value to an integer.

Reply:
Ok, I see why the BASE is necessary here. Sorry for the spam. Still doesn't work though.

Reply:
Adding a check to see if we can even add another digit would address this:
```cpp  
template <int Base, class T>      
bool add_ascii(T& x, char c, enable_if_tt<std::is_integral<T>, int> u = 0) {      
  CAF_IGNORE_UNUSED(u);      
  // Check if we can add another digit.      
  T res = (x * Base);                                          
  if (res / Base != x)                                         
    return false;                                           
  // Check if we can add this specific digit.              
  ascii_to_int<Base, T> f;                                                                
  auto y = f(c);                                                                                  
  if (x > ((std::numeric_limits<T>::max() / Base) - y)                                                           
            + (std::numeric_limits<T>::max() % Base))                                                            
    return false;                                                                                                
  x = res + y;                                                                                                   
  return true;                                                                                                   
}
```